html/user/am_set_info.php

```diff
@@ -210,6 +222,26 @@ function success($x) {
     $query .= " passwd_hash='$database_passwd_hash', ";
 }
 
+// Build the query for the consent table. consent_id must be present.
+$cquery = "";
+$now = time();
+if ($consent_id) {
+    $cquery .= "consent_id=$consent_id";
+    if ($consent_settime==1) {
+        $cquery .= ",consent_time=$now";
+    }
```

Comment:
I think it would be better if `consent_time` is set any time consent is given or changed and account managers have no say in the matter.

As for the rest of the consent parameters I would check that either they are all set or none of them are set. I don't think setting only some of them would work well.

Reply:
I agree with you in principle, but I received very little feedback from AM developers on what would be best for them to be able to change, hence a very flexible/open API for interaction.